spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -252,20 +258,28 @@ object DistriOptimizer {
           }
 
           parameters.putGradients(cached.gradient)
-          tasks ++= Engine.default.invoke((0 until _subModelNumber).map(i => () => {
-            cached.localModels(i).training()
-            cached.localModels(i).zeroGradParameters()
-          }))
+          tasks ++= Engine.default.invoke {
+            (0 until _subModelNumber).map { i =>
+              () => {
+                cached.localModels(i).training()
+                cached.localModels(i).zeroGradParameters()
+              }
+            }
+          }
           Iterator.single(finishedThreads.size)
-        }).reduce(_ + _)
-
-      dropModelNumBatch += (driverSubModelNum - finishedModelNum)
-      if (dropPercentage == 0 || finishedModelNum >= driverSubModelNum * (1-maxDropPercentage)) {
-        val value = lossSum.value / finishedModelNum
-        models.mapPartitions(modelIter => {
+        }
+      }
+      val numFinishedModelUpdates = finishedThreads.reduce(_ + _)
```

Comment:
Update code like val numFinishedModel = dataRDD.zipPartitoins(xxxxx).reduce(_ + _), similar to original code? Currently looks like there are two finishedThread variables which have different meaning and different type, which kind of cause code less clear.